nexus/types/src/deployment.rs

```diff
@@ -998,6 +1035,12 @@ pub struct BlueprintMetadata {
     pub internal_dns_version: Generation,
     /// external DNS version when this blueprint was created
     pub external_dns_version: Generation,
+    /// CockroachDB state fingerprint when this blueprint was created
+    pub cockroachdb_fingerprint: String,
+    /// Whether to set `cluster.preserve_downgrade_option` and what to set it to
+    /// (`None` if this value was retrieved from the database and was invalid)
+    pub cockroachdb_setting_preserve_downgrade:
+        Option<CockroachDbPreserveDowngrade>,
```

Comment:
Small nit, feel free to ignore - do you think this would be clearer as a `Result<CockroachDbPreserveDowngrade, _>` where the error case indicates "something is wrong in the DB" and probably holds the actual string we couldn't deserialize?

Reply:
Hmm, so a `Result<CockroachDbPreserveDowngrade, String>`, and then handle that in the blueprint diff spots? idk if that would be clearer but it might be nicer regardless.

Reply:
Yeah, or maybe a newtype around a string (for clarity of what that string is). I don't feel strongly about this either way.